crates/re_log_types/src/data_cell.rs

```diff
@@ -0,0 +1,445 @@
+    #[inline]
+    pub fn as_arrow(&self) -> Box<dyn arrow2::array::Array> {
+        self.values.clone() /* shallow */
+    }
+
+    /// Returns the contents of the cell as a reference to an arrow array.
+    ///
+    /// Avoid using raw arrow arrays unless you absolutely have to.
+    /// If you do use them, try to keep the scope as short as possible: holding on to a raw array
+    /// might prevent the datastore from releasing memory from garbage collected data.
+    #[inline]
+    pub fn as_arrow_ref(&self) -> &dyn arrow2::array::Array {
+        &*self.values
+    }
+
+    /// Returns the contents of the cell as an arrow array (shallow clone) wrapped in a unit-length
+    /// list-array.
+    ///
+    /// Useful when dealing with cells of different lengths in context that don't allow for it.
+    ///
+    /// * Before: `[C, C, C, ...]`
+    /// * After: `ListArray[ [C, C, C, C] ]`
+    //
+    // TODO(#1696): this shouldn't be public, need to make it private once the store has been
+    // patched to use datacells directly.
+    // TODO(cmc): effectively, this returns a `DataColumn`... think about that.
+    #[doc(hidden)]
+    #[inline]
+    pub fn as_arrow_monolist(&self) -> Box<dyn arrow2::array::Array> {
+        use arrow2::{array::ListArray, offset::Offsets};
+
+        let values = self.as_arrow();
+        let datatype = self.datatype().clone();
+
+        let datatype = ListArray::<i32>::default_datatype(datatype);
+        let offsets = Offsets::try_from_lengths(std::iter::once(self.num_instances() as usize))
+            .unwrap()
+            .into();
+        let validity = None;
+
+        ListArray::<i32>::new(datatype, offsets, values, validity).boxed()
+    }
+
+    /// Returns the contents of the cell as an iterator of native components.
+    ///
+    /// Fails if the underlying arrow data cannot be deserialized into `C`.
+    //
+    // TODO(#1694): There shouldn't need to be HRTBs here.
+    #[inline]
+    pub fn try_as_native<C: DeserializableComponent>(
+        &self,
+    ) -> DataCellResult<impl Iterator<Item = C> + '_>
+    where
+        for<'a> &'a C::ArrayType: IntoIterator,
+    {
+        use arrow2_convert::deserialize::arrow_array_deserialize_iterator;
+        arrow_array_deserialize_iterator(&*self.values).map_err(Into::into)
+    }
+
+    /// Returns the contents of the cell as an iterator of native components.
+    ///
+    /// Panics if the underlying arrow data cannot be deserialized into `C`.
+    /// See [`Self::try_as_native`] for a fallible alternative.
+    //
+    // TODO(#1694): There shouldn't need to be HRTBs here.
+    #[inline]
+    pub fn as_native<C: DeserializableComponent>(&self) -> impl Iterator<Item = C> + '_
+    where
+        for<'a> &'a C::ArrayType: IntoIterator,
+    {
+        self.try_as_native().unwrap()
+    }
+}
+
+impl DataCell {
+    /// The name of the component type stored in the cell.
+    #[inline]
+    pub fn component(&self) -> ComponentName {
```

Comment:
I think it would be nice to be explicit here and call this `component_name`

Reply:
agreed